src/BookReader.js

```diff
@@ -662,87 +671,140 @@ BookReader.prototype.resize = function() {
 };
 
 /**
- * Binds keyboard event listeners
+ * Returns true if passed element activates or maintains BookReader keyboard shortcut focus
+ * @param {HTMLElement} element
+ * @returns {boolean}
  */
-BookReader.prototype.setupKeyListeners = function() {
-  var self = this;
+BookReader.prototype.isKeyFocus = function (element) {
+  // BookReader and ItemNavigator
+  let selectors = this.el;
+  selectors += ',' + 'item-navigator';
+  // Top level elements
+  // NOTE: a <nav> element containing both
+  // or a standard banner container (<ia-bannners>?)
+  // would be a helpful design change here
+  selectors += ',' + '#topnav';
+  selectors += ',' + '#donate_banner';
+  return Boolean(element.closest(selectors));
+};
 
-  var KEY_PGUP = 33;
-  var KEY_PGDOWN = 34;
-  var KEY_END = 35;
-  var KEY_HOME = 36;
-
-  var KEY_LEFT = 37;
-  var KEY_UP = 38;
-  var KEY_RIGHT = 39;
-  var KEY_DOWN = 40;
-  // The minus(-) and equal(=) keys have different mappings for different browsers
-  var KEY_MINUS = 189; // Chrome
-  var KEY_MINUS_F = 173; // Firefox
-  var KEY_NUMPAD_SUBTRACT = 109;
-  var KEY_EQUAL = 187; // Chrome
-  var KEY_EQUAL_F = 61; // Firefox
-  var KEY_NUMPAD_ADD = 107;
-
-  // We use document here instead of window to avoid a bug in jQuery on IE7
-  $(document).on("keydown", function(e) {
-
-    // Keyboard navigation
-    switch (e.keyCode) {
-    case KEY_PGUP:
-    case KEY_UP:
-      // In 1up mode page scrolling is handled by browser
-      if (!utils.isInputActive() && self.constMode2up == self.mode) {
-        e.preventDefault();
-        self.prev();
+/**
+ * Binds keyboard and keyboard focus event listeners
+ */
+BookReader.prototype.setupKeyListeners = function () {
+
+  // Keyboard focus by click location
+  document.addEventListener('click', (event) => {
+    this.hasKeyFocus = this.isKeyFocus(event.target);
+  });
+
+  // Keyboard focus by tabbed element location
+  document.addEventListener('focusin', (event) => {
+    this.hasKeyFocus = this.isKeyFocus(event.target);
+  });
+
+  // Keyboard focus by BookReader in viewport
+  //
+  // Intersection observer and callback sets BookReader keyboard
+  // "focus" flag off when the BookReader is not in the viewport.
+  const observer = new IntersectionObserver((entries) => {
+    entries.forEach((entry) => {
+      if (entry.intersectionRatio === 0) {
+        this.hasKeyFocus = false;
       }
+    });
+  }, {
+    root: null,
+    rootMargin: '0px',
+    threshold: [0, 1],
+  });
+  observer.observe(this.refs.$br);
```

Comment:
Looking at the CI output: https://github.com/internetarchive/bookreader/runs/4465817244?check_suite_focus=true#step:10:193

It looks like this is what's causing it to error; `$br` is a jquery el! Need to convert it to an HTMLElement

```suggestion
  observer.observe(this.refs.$br[0]);
```

Reply:
Thanks @cdrini -- already caught and fixed this